### PR TITLE
fix: update yarn packages, prettier

### LINKED ISSUE #516: `prettier` in `package.json` calling `forge fmt`
I just realized that this could be causing the discrepancy with the formatting. Because this task is named "prettier" it is possible that lint-staged is calling this instead of what we expect.

_Originally posted by @pavlovcik in https://github.com/ubiquity/ubiquity-dollar/pull/417#discussion_r1085080563_

---
```
 "devDependencies": {
    "cspell": "latest",
    "prettier": "latest",
    "prettier-plugin-solidity": "latest"
  }
```

if we agree to keep prettier over forge fmt we need to :
1 - add the dev dependencies as stated above 
2 - update the overrides section of the  .prettierrc in the root folder

_Originally posted by @zgorizzo69 in  https://github.com/ubiquity/ubiquity-dollar/pull/417#issuecomment-1400977221_

### PR DESCRIPTION
**Pull request by molecula451**
Resolves #516 

This PR pass conventional commits, and upgrades & locks requested yarn packages on the issue, this PR does not override prettier config for any of other files than initially was sice it's not clarified which other files we want to prettify, request changes accordingly

### CODE REVIEW

**Comment by 0x4007** [comment]
Not sure if "latest" is valid and you didn't finish implementing the spec. Lastly, unfortunately the rules are that we allow for 7 days of inactivity before allowing another bounty hunter to submit their solution. As we have more developers in the DevPool we may consider shortening this time period.

**Comment by molecula451** [comment]
> Not sure if "latest" is valid and you didn't finish implementing the spec.

I update to latest as the originally posted on the issue, you may request the changes

```
 "devDependencies": {
    "cspell": "latest",
    "prettier": "latest",
    "prettier-plugin-solidity": "latest"
  }
```

https://github.com/ubiquity/ubiquity-dollar/issues/516

Yes, I agree, it's just looked like the issue was not moving forward, just let me know!

>Lastly, unfortunately the rules are that we allow for 7 days of inactivity before allowing another bounty hunter to submit their solution. As we have more developers in the DevPool we may consider shortening this time period.

**Comment by 0x4007** [comment]
> Yes, I agree, it's just looked like the issue was not moving forward, just let me know!

I appreciate your eagerness to close issues out. Tag me in a couple of days and I can credit you if the active assignee neglects to finish.

**Comment by molecula451** [comment]
> Not sure if "latest" is valid and you didn't finish implementing the spec. Lastly, unfortunately the rules are that we allow for 7 days of inactivity before allowing another bounty hunter to submit their solution. As we have more developers in the DevPool we may consider shortening this time period.

yes "latest" it's a valid tag for npm/yarn packages

**Review by 0x4007 — COMMENTED** [review]
This seems correct so far

**Comment by 0x4007** [comment]
You need to rename the "prettier" script to "format" in the `package.json` and address the bit about `.prettierrc` comment in the specification. @rndquu do you think you can make sure the spec is followed with @molecula451's implementation?

**Comment by molecula451** [comment]
provide more context for the `.prettierrc` file @rndquu input would be great

**Comment by 0x4007** [comment]
> provide more context for the `.prettierrc` file @rndquu input would be great

> 2 - update the overrides section of the .prettierrc in the root folder

**Comment by molecula451** [comment]
> > provide more context for the `.prettierrc` file @rndquu input would be great
> 
> > 2 - update the overrides section of the .prettierrc in the root folder

`.prettierrc` may handle different styles according to a specific taste in prettifying the files, suggest you guys agree on one so I shall commit it

**Comment by rndquu** [comment]
We use prettier as a solidity linter. Lint-staged also calls prettier on commit.

The original [issue](https://github.com/ubiquity/ubiquity-dollar/issues/516) describes a moment in time when we used `forge fmt` as a linter but we moved to prettier so this [issue](https://github.com/ubiquity/ubiquity-dollar/issues/516) should have been closed as "not planned".

Regarding this PR: we shouldn't use `latest` because it introduces dependencies updated with **major versions** which may contain breaking changes. 

So this PR basically brings us:
1. script rename `prettier:all => format:all`
2. some foundry config [optimisations](https://github.com/ubiquity/ubiquity-dollar/pull/685/files#diff-5084a3d2809ef6c86199f46d1d985964c92c3a98247da067d6cb04b18b0a40a9L18)

We can either close this issue as "not planned" either merge this PR (but remove the `latest` versions in `package.json`).

**Comment by molecula451** [comment]
> 

Why not planned? you guys agree on a prettier style and we'll do it

**Comment by rndquu** [comment]
> > 
> 
> Why not planned? you guys agree on a prettier style and we'll do it

I don't see any issues with the current prettier setup for solidity

**Comment by molecula451** [comment]
> > > 
> > 
> > 
> > Why not planned? you guys agree on a prettier style and we'll do it
> 
> I don't see any issues with the current prettier setup for solidity

i don't either, @pavlovcik you may leave the current prettier config, you may propose a change later, this PR is done

Yeah, updating the packages as rndquu request

**Comment by rndquu** [comment]
> > > > 
> > > 
> > > 
> > > Why not planned? you guys agree on a prettier style and we'll do it
> > 
> > 
> > I don't see any issues with the current prettier setup for solidity
> 
> i don't either, @pavlovcik you may leave the current prettier config, you may propose a change later, this PR is done
> 
> Yeah, updating the packages as rndquu request

pls also update this [one](https://github.com/ubiquity/ubiquity-dollar/pull/685/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R18)

**Comment by molecula451** [comment]
> > > > > 
> > > > 
> > > > 
> > > > Why not planned? you guys agree on a prettier style and we'll do it
> > > 
> > > 
> > > I don't see any issues with the current prettier setup for solidity
> > 
> > 
> > i don't either, @pavlovcik you may leave the current prettier config, you may propose a change later, this PR is done
> > Yeah, updating the packages as rndquu request
> 
> pls also update this [one](https://github.com/ubiquity/ubiquity-dollar/pull/685/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R18)

bumped https://github.com/ubiquity/ubiquity-dollar/pull/685/commits/2aff1a80da17624f87e8325cc9b5b9ce2c560678

**Review by rndquu — APPROVED** [review]

**Comment by molecula451** [comment]
let's finish this one

**Comment by rndquu** [comment]
@molecula451 could you resolve code conflicts?

**Comment by molecula451** [comment]
> @molecula451 could you resolve code conflicts?

ther you go, https://github.com/ubiquity/ubiquity-dollar/pull/685/commits/c7f3bcdbf718622810fae2478662731abf1ed522

please merge to avoid code conflicts on the branch

**Comment by molecula451** [comment]
@rndquu i just re-synced your PR into the branch

File: package.json [hunk 1]
```diff
@@ -34,8 +34,9 @@
     "dev:dapp": "yarn workspace @ubiquity/dapp run start",
     "test:all": "yarn workspace @ubiquity/contracts run test:unit",
     "coverage": "yarn workspace @ubiquity/contracts run test:coverage",
+    "format:all": "prettier packages/ --write",
+    "purge:all": "yarn workspaces foreach run purge && rimraf node_modules",
     "clean:all": "yarn workspaces foreach run clean && rimraf node_modules",
```
Comment (rndquu) [inline package.json:38]: `clean:all` and `purge:all` are equal, pls leave only the `clean:all` command

**Review by rndquu — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (molecula451) [reply package.json:38]: updated here https://github.com/ubiquity/ubiquity-dollar/pull/685/commits/aca85372e08e7124b8627a8810f794104ee699e2

**Review by rndquu — APPROVED** [review]

**Comment by 0x4007** [comment]
> i don't either, @pavlovcik you may leave the current prettier config, you may propose a change later, this PR is done

We use `prettier` for everything, including for the UI code under `packages/dapp/` and other codebases. Prettier can also format the Solidity code so the Forge formatter is not necessary.

**Review by 0x4007 — APPROVED** [review]

**Pull request merged**